### ISSUE TITLE
Possible bug

### ISSUE THREAD

**Issue by tatp22** [tatp22/linformer-pytorch#8]
I may have discovered a possible bug. When I run `python pretrain_tutorial.py`, I get some vertical lines when running the visualizer on the random data (Try it yourself, and run the trained model on a new random array).

In the extreme case, I get that all of the queries are attending to the same key. This leads to the effect that, for an input of size `(batch_size, input_len, ch)`, every vector of length `ch` on the `input_len` axis will have the same value. To put this in a concrete example, imagine a 32x32x3 picture (RGB, as an example), fed in the model of a batch size of `1`,. If the input is `(1,32*32,3)`, every channel of the image will be the same value on every pixel. For example, the R channel will have the value of `128` on each channel, the G channel might have `43`, and the B channel may have `212`. If the input is `(1,3,32*32)`, every R,G,B channel will look like an image, but they will have the same exact pixel values.

The problem is being investigated. I believe it is a problem with the MHAttention layer, or possibly the positional embedding, but I cannot say for sure. Will update when a fix is found.

**Comment by tatp22** [comment]
I believe the bug has been found. This was in fact not a problem with the `MHAttention` class, but rather in the way the modules were sequenced. For some reason, the way that I previously indexed my models led to this phenomenon. I'm not exactly how this fixed the problem, but it did, and I'll be pushing a new version shortly (Actually, should already be up in `0.12.4`.

Thank you again to [lucidrains](https://github.com/lucidrains) for coming up with previous transformer implementations so I can take inspiration from them.

**Comment by lucidrains** [comment]
@tatp22 oh no problem Peter! :)
